ui/v2.5/src/components/Images/ImageWall.tsx

```diff
@@ -0,0 +1,191 @@
+  faPlus,
+  faTag,
+} from "@fortawesome/free-solid-svg-icons";
+import { Icon, TagLink, HoverPopover, SweatDrops } from "src/components/Shared";
+import * as GQL from "src/core/generated-graphql";
+import { objectTitle } from "src/core/files";
+import { Link } from "react-router-dom";
+
+interface IImageWallProps {
+  images: GQL.SlimImageDataFragment[];
+  onChangePage: (page: number) => void;
+  currentPage: number;
+  pageCount: number;
+  handleImageOpen: (index: number) => void;
+}
+
+export const ImageWall: React.FC<IImageWallProps> = ({
+  images,
+  handleImageOpen,
+}) => {
+  const thumbs = images.map((image, index) => {
+    function maybeRenderTagPopoverButton() {
+      if (image.tags.length <= 0) return;
+
+      const popoverContent = image.tags.map((tag) => (
+        <TagLink key={tag.id} tag={tag} tagType="image" />
+      ));
+
+      return (
+        <HoverPopover
+          className="tag-count"
+          placement="bottom"
+          content={popoverContent}
+        >
+          <Button className="minimal">
+            <Icon icon={faTag} />
+            <span>{image.tags.length}</span>
+          </Button>
+        </HoverPopover>
+      );
+    }
+
+    function maybeRenderPerformerPopoverButton() {
+      if (image.performers.length <= 0) return;
+
+      return <PerformerPopoverButton performers={image.performers} />;
+    }
+
+    function maybeRenderOCounter() {
+      if (image.o_counter) {
+        return (
+          <div className="o-count">
+            <Button className="minimal">
+              <span className="fa-icon">
+                <SweatDrops />
+              </span>
+              <span>{image.o_counter}</span>
+            </Button>
+          </div>
+        );
+      }
+    }
+
+    function maybeRenderGallery() {
+      if (image.galleries.length <= 0) return;
+
+      const popoverContent = image.galleries.map((gallery) => (
+        <TagLink key={gallery.id} gallery={gallery} />
+      ));
+
+      return (
+        <HoverPopover
+          className="gallery-count"
+          placement="bottom"
+          content={popoverContent}
+        >
+          <Button className="minimal">
+            <Icon icon={faImages} />
+            <span>{image.galleries.length}</span>
+          </Button>
+        </HoverPopover>
+      );
+    }
+
+    function maybeRenderOrganized() {
+      if (image.organized) {
+        return (
+          <div className="organized">
+            <Button className="minimal">
+              <Icon icon={faBox} />
+            </Button>
+          </div>
+        );
+      }
+    }
+
+    function renderEdit() {
+      return (
+        <div className="edit">
+          <Button className="minimal">
+            <Icon icon={faPlus} />
+          </Button>
+        </div>
+      );
+    }
+    const ref = useRef(null);
+    const handleTouchEnd = (event) => {
+      if (ref.current === document.activeElement) {
+        alert("If i would to display this, it would be nice!");
```

Comment:
Here is where the trouble is.
I'm trying to use the ref.current that points to my element (i checked it does), and focus().
But it doesnt seems to happen. If i do
`ref.current.focus()` and then check `ref.current === document.activeElement` its always false. This means the element is never focused.

Reply:
Solved by @InfiniteTF